zebra-network/src/protocol/external/tests/preallocate.rs

```diff
@@ -1,28 +1,37 @@
 //! Tests for trusted preallocation during deserialization.
 
-use super::super::inv::{InventoryHash, INV_HASH_SIZE};
+use super::super::inv::InventoryHash;
 
 use zebra_chain::serialization::{TrustedPreallocate, ZcashSerialize, MAX_PROTOCOL_MESSAGE_LEN};
 
 use proptest::prelude::*;
 use std::convert::TryInto;
 
 proptest! {
-    /// Confirm that each InventoryHash takes exactly INV_HASH_SIZE bytes when serialized.
-    /// This verifies that our calculated [`TrustedPreallocate::max_allocation`] is indeed an upper bound.
+    /// Confirm that each InventoryHash takes the expected size in bytes when serialized.
     #[test]
-    fn inv_hash_size_is_correct(inv in InventoryHash::arbitrary()) {
+    fn inv_hash_size_is_correct(inv in any::<InventoryHash>()) {
         let serialized_inv = inv
             .zcash_serialize_to_vec()
             .expect("Serialization to vec must succeed");
-        assert!(serialized_inv.len() == INV_HASH_SIZE);
+
+        let expected_size = match inv {
+            InventoryHash::Error
+            | InventoryHash::Tx(_)
+            | InventoryHash::Block(_)
+            | InventoryHash::FilteredBlock(_) => 32 + 4,
+
+            InventoryHash::Wtx(_) => 32 + 32 + 4,
+        };
+
+        assert_eq!(serialized_inv.len(), expected_size);
     }
 
     /// Verifies that...
     /// 1. The smallest disallowed vector of `InventoryHash`s is too large to fit in a legal Zcash message
     /// 2. The largest allowed vector is small enough to fit in a legal Zcash message
     #[test]
-    fn inv_hash_max_allocation_is_correct(inv in InventoryHash::arbitrary()) {
+    fn inv_hash_max_allocation_is_correct(inv in InventoryHash::smallest_types_strategy()) {
```

Comment:
Future Bug Risk: It might be better to generate all variants, then skip any known larger ones using `prop_assume!`.

That way, if we add an even smaller variant, but forget to update `smallest_types_strategy`, this test will fail.

I wouldn't worry about skipping too many runs here, this test basically only tests the serialized size.